setroot.c

```diff
@@ -955,6 +995,26 @@ parse_opts( unsigned int argc, char **args )
             aspect = FIT_HEIGHT;
         } else if   (streq(token, "-fw") || streq(token, "--fit-width")) {
             aspect = FIT_WIDTH;
+        } else if   (streq(token, "-gx") || streq(token, "--gravity-x")) {
+            int last = strlen(args[++i]) - 1;
+            if (args[i][last] == '%') {
```

Comment:
`args` are null-terminated so this check will never pass; you would need to check `last - 1`